src/mscorlib/src/System/AppContext/AppContext.cs

```diff
@@ -65,79 +64,83 @@ public static bool TryGetSwitch(string switchName, out bool isEnabled)
+                // We get the value of isEnabled from the value that we stored in the dictionary
+                isEnabled = (switchValue & SwitchValueState.HasTrueValue) == SwitchValueState.HasTrueValue; 
 
+                // 2. The switch has a valid value AND we have checked for overrides
+                if ((switchValue & SwitchValueState.HasLookedForOverride) == SwitchValueState.HasLookedForOverride)
+                {
                     return true;
                 }
-                else
+
+                // 3. The switch has a valid value, but we need to check for overrides.
+                // Regardless of whether or not the switch has an override, we need to update the value to reflect
+                // the fact that we checked for overrides. 
+                bool overrideValue;
+                if (AppContextDefaultValues.TryGetSwitchOverride(switchName, out overrideValue))
                 {
-                    // The value is NOT in the dictionary
-                    // In this case we need to see if we have an override defined for the value.
-                    // There are 2 cases:
-                    // 1. The value has an override specified. In this case we need to add the value to the dictionary 
-                    //    and mark it as checked for overrides
-                    //    Example: In a servicing event, System.Xml introduces a switch and an override is specified.
-                    //             The value is not found in mscorlib (as System.Xml ships independent of mscorlib)
-                    // 2. The value does not have an override specified
-                    //    In this case, we want to capture the fact that we looked for a value and found nothing by adding 
-                    //    an entry in the dictionary with the 'sentinel' value of 'SwitchValueState.UnknownValue'.
-                    //    Example: This will prevent us from trying to find overrides for values that we don't have in the dictionary
-
-                    // 1. The value has an override specified.
-                    bool overrideValue;
-                    if (AppContextDefaultValues.TryGetSwitchOverride(switchName, out overrideValue))
-                    {
-                        isEnabled = overrideValue;
-
-                        // Update the switch in the dictionary to mark it as 'checked for override'
-                        s_switchMap[switchName] = (isEnabled ? SwitchValueState.HasTrueValue : SwitchValueState.HasFalseValue)
-                                                    | SwitchValueState.HasLookedForOverride;
-
-                        return true;
-                    }
+                    // we found an override!
+                    isEnabled = overrideValue;
+                }
+
+                // Update the switch in the dictionary to mark it as 'checked for override'
+                SetSwitch(switchName, isEnabled);
```

Comment:
I would just explicitly set the value here. Calling SetSwitch is going to go through a lot more checks and code before it sets the value.